pygmt/clib/session.py

```diff
@@ -2066,6 +2079,17 @@ def virtualfile_to_raster(
                 self["GMT_IS_IMAGE"]: "image",
                 self["GMT_IS_CUBE"]: "cube",
             }[family]
+
+        if kind == "image":  # Use temporary file for images
```

Comment:
This workaround can be removed after PR #3128, then rioxarray is no longer required for writing images.